Iwasawalib/RingTheory/PseudoNull/StructureTheorem.lean

```diff
@@ -39,12 +40,107 @@ class IsKrullDomain (A : Type*) [CommRing A] [IsDomain A] : Prop where
 
 instance (priority := 100) IsKrullDomain.isIntegrallyClosed
     (A : Type*) [CommRing A] [IsDomain A] [IsKrullDomain A] : IsIntegrallyClosed A := by
-  sorry
+  have notField : ∀ (p : PrimeSpectrum A), p.1.primeHeight = 1 → ¬IsField (Localization p.1.primeCompl) := by
```

Comment:
Could you add this theorem
```lean
-- will be in mathlib soon
theorem IsIntegrallyClosed.of_localization {R : Type*} [CommRing R] [IsDomain R]
    (S : Set (PrimeSpectrum R)) (h : ∀ p ∈ S, IsIntegrallyClosed (Localization.AtPrime p.1))
    (hs : ⨅ p ∈ S, (Localization.subalgebra (FractionRing R) p.1.primeCompl
      p.1.primeCompl_le_nonZeroDivisors) = ⊥) : IsIntegrallyClosed R := by
  sorry
```
before and prove this result using `IsIntegrallyClosed.of_localization`?